data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java

```diff
@@ -122,9 +127,14 @@ public OpenTelemetrySdk setup() {
     sdkBuilder.setTracerProvider(
       tracerProviderBuilder.build()
     );
-    sdkBuilder.setPropagators(ContextPropagators.create(
-      W3CTraceContextPropagator.getInstance()
-    ));
+
+    ContextPropagators contextPropagators;
+    switch (getHeadsFormat()) {
+      case B3_MULTI_HEADER -> contextPropagators = ContextPropagators.create(B3Propagator.injectingMultiHeaders());
+      case B3_SINGLE_HEADER -> contextPropagators = ContextPropagators.create(B3Propagator.injectingSingleHeader());
+      default -> contextPropagators = ContextPropagators.create(W3CTraceContextPropagator.getInstance());
+    }
```

Comment:
This looks a little bit easier to read:
```suggestion
    final var contextPropagators = switch (getHeadsFormat()) {
      case B3_MULTI_HEADER -> ContextPropagators.create(B3Propagator.injectingMultiHeaders());
      case B3_SINGLE_HEADER -> ContextPropagators.create(B3Propagator.injectingSingleHeader());
      default -> ContextPropagators.create(W3CTraceContextPropagator.getInstance());
    };
```